### PR TITLE
Optimizing logs

### PR DESCRIPTION
**Pull request by dinusha92**
## Purpose
Optimizing logs for the ease of debugging.

## Goals
Introduce fullError, debug methods.

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/gateway/file_rotating_timer.bal [hunk 1]
```diff
@@ -46,7 +46,7 @@ function sendFileRotatingEvent() returns error? {
 }
 
 function errorOnRotating(error e) {
-    log:printDebug("File not present to rotate:" + e.message);
+    printDebug(KEY_ROTATE_TASK,"File not present to rotate:" + e.message);
```
Comment (chamilaadhi) [inline components/micro-gateway-core/src/main/ballerina/gateway/file_rotating_timer.bal]: format

File: components/micro-gateway-core/src/main/ballerina/gateway/data_writer.bal [hunk 2]
```diff
@@ -133,10 +133,11 @@ function writeEventToFile(EventDTO eventDTO) {
     } finally {
         match charChannel.close() {
             error sourceCloseError => {
-                log:printError("Error occurred while closing the channel: ", err = sourceCloseError);
+                printError(KEY_ANALYTICS_FILTER, "Error occurred while closing the channel: " + sourceCloseError.message
+                );
```
Comment (chamilaadhi) [inline components/micro-gateway-core/src/main/ballerina/gateway/data_writer.bal]: format

Comment (dinusha92) [reply components/micro-gateway-core/src/main/ballerina/gateway/data_writer.bal]: done

File: components/micro-gateway-core/src/main/ballerina/gateway/file_rotating_timer.bal (hunk 1 above)
Comment (dinusha92) [reply components/micro-gateway-core/src/main/ballerina/gateway/file_rotating_timer.bal]: done

**Pull request merged**